### PR TITLE
Atlas 2787

### PR DESCRIPTION
**Pull request by BennuAshes**

### CODE REVIEW

**Review by nthornton2010 — CHANGES_REQUESTED** [review]

File: packages/cdk-routing-split/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "package": "npm run build && serverless package --stage sandbox",
         "deploy:sandbox": "npm run build && serverless deploy --stage sandbox --aws-profile sandbox --verbose",
         "test:sandbox": "sls export-env --stage sandbox --aws-profile sandbox && jest",
+        "test:watchsandbox": "sls export-env --stage sandbox --aws-profile sandbox && jest --watch",
```
[the remark this answers is not in the document]
Comment (BennuAshes) [reply packages/cdk-routing-split/package.json:11]: i think ill just get rid of it, i ended up not using it that much. i think your suggestion of -- --watch is good

File: packages/cdk-routing-split/src/get-origin/handler.ts [hunk 2]
```diff
@@ -1,25 +1,60 @@
...
         logger.debug('get-origin event', { event });
 
-        return {
-            statusCode: 200,
-            body: action(),
+        const request: CloudFrontRequest = event.Records[0].cf.request;
+        const { domain, subdomain } = parse(request.origin.custom.domainName);
+        const pathname = getPathnameSegment(request.uri);
+        const queryStrings = getQueryStrings(request.querystring);
+        console.log('jest-', subdomain, domain, pathname, queryStrings);
```
[the remark this answers is not in the document]
Comment (BennuAshes) [reply packages/cdk-routing-split/src/get-origin/handler.ts]: i don't see this anymore, im fairly certain i removed it as part of the update lambda (that got merged last night)

File: packages/cdk-routing-split/test/cases/unit/get-origin.test.ts [hunk 3]
```diff
@@ -1,9 +1,40 @@
...
+                },
+            ],
+        },
+        status: '404',
+        statusDescription: 'Not Found',
+    };
+    beforeEach(() => {
+        request = given.a_getOrigin_event();
+    });
+    it('calls get-origin with no domainName', async () => {
```
[the remark this answers is not in the document]
Comment (BennuAshes) [reply packages/cdk-routing-split/test/cases/unit/get-origin.test.ts:22]: ill change it, its bugging me now that you pointed it out

File: packages/cdk-routing-split/test/cases/integration/get-origin.test.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        addRequest = given.an_add_request_body();
+        domainData = getDomainData(url);
+        originRequest = given.a_getOrigin_event();
+        originRequest.Records[0].cf.request.origin.custom.domainName = `${domainData.subdomain}${domainData.subdomain ? '.' : ''}${
+            domainData.domain
+        }`;
+        originRequest.Records[0].cf.request.querystring = domainData.queryStrings.join('&');
+        originRequest.Records[0].cf.request.uri = domainData.pathname;
+    });
+    it('calls get-origin with a valid url with a subdomain', async () => {
```
[the remark this answers is not in the document]
Comment (BennuAshes) [reply packages/cdk-routing-split/test/cases/integration/get-origin.test.ts:24]: Ah yea great idea, adding it now.

File: packages/cdk-routing-split/test/cases/integration/get-origin.test.ts [hunk 5]
```diff
@@ -125,4 +125,60 @@ describe('When an entity', () => {
...
+
+        const subdomainRequest = {
+            key: UrlSegment.Subdomain,
+            value: domainData.subdomain,
+            priority: 3,
+            origin: given.a_simple_url(),
+        } as AddRequest;
+        await when.we_invoke_add(subdomainRequest);
+
+        const origin = pathRequest.origin;
```
Comment (nthornton2010) [inline packages/cdk-routing-split/test/cases/integration/get-origin.test.ts:162]: This should be domain request. The lowest number has the highest priority, so the domainRequest's origin is what should be expected.

File: packages/cdk-routing-split/test/setup/jest.setup.js [hunk 6]
```diff
@@ -9,8 +9,8 @@ if (process.env.DSI_PARAMETER_NAME && process.env.DSI_PARAMETER_NAME.includes('s
 
 // prevent lambdas from logging to jest
 global.console = {
-    log: jest.fn(),
-    //log: console.log, //Uncomment for debugging purposes
+    //log: jest.fn(),
```
Comment (nthornton2010) [inline packages/cdk-routing-split/test/setup/jest.setup.js:12]: This shouldn't have been committed.

File: packages/cdk-routing-split/test/steps/when.ts [hunk 7]
```diff
@@ -9,6 +9,7 @@ import { Request as DeleteRequest } from '../../src/delete/request';
 import { Request as UpdateRequest } from '../../src/update/request';
 
 export const we_invoke_add = async (body: AddRequest) => {
+    console.log('add - body', body);
```
Comment (nthornton2010) [inline packages/cdk-routing-split/test/steps/when.ts:12]: A new console.log was added. Please make sure you're code reviewing your changes before committing them.

**Review by nthornton2010 — CHANGES_REQUESTED** [review]

File: packages/cdk-routing-split/test/steps/when.ts (hunk 7 above)
Comment (nthornton2010) [reply packages/cdk-routing-split/test/steps/when.ts:12]: Oh, this is a test file, sorry my mistake.

**Pull request merged**